docker-compose.yaml

```diff
@@ -1,36 +1,17 @@
-version: "3.7"
-
 services:
   caddy:
     container_name: caddy
     image: docker.io/library/caddy:2-alpine
     network_mode: host
     volumes:
       - ./Caddyfile:/etc/caddy/Caddyfile:ro
-      - caddy-data:/data:rw
-      - caddy-config:/config:rw
+      - caddy-data:/data
+      - caddy-config:/config
     environment:
       - SEARXNG_HOSTNAME=${SEARXNG_HOSTNAME:-http://localhost:80}
       - SEARXNG_TLS=${LETSENCRYPT_EMAIL:-internal}
-    cap_drop:
-      - ALL
-    cap_add:
-      - NET_BIND_SERVICE
-
-  redis:
-    container_name: redis
-    image: docker.io/library/redis:alpine
-    command: redis-server --save 30 1 --loglevel warning
-    networks:
-      - searxng
-    volumes:
-      - redis-data:/data
-    cap_drop:
-      - ALL
-    cap_add:
-      - SETGID
-      - SETUID
-      - DAC_OVERRIDE
+    security_opt:
+      - no-new-privileges:true
```

Comment:
I don't think `no-new-privileges` drop all the default privileges.

Reply:
No, it doesn't, but it prevents permission escalation inside the container. Further limiting the capabilities with `CAP_DROP` might be a good idea if you keep up with what you intend to run, but in this case at any time new capabilities may be required (not only for SearXNG, but for Valkey or Caddy) and the execution of compose could fail.

Reply:
I don't get it, it has always worked fine with cap_drop? Why is the execution of compose failing now?

Reply:
Not that it fails now, but it may fail later. However, if you would like to keep that functionality I'll add it back.

Reply:
This hasn't failed for 5 years so I prefer to not touch it. Thanks

Reply:
Can you remove `no-new-privileges:true` because it does the same as "cap-drop": https://raesene.github.io/blog/2019/06/01/docker-capabilities-and-no-new-privs/